### ISSUE TITLE
Ignore Hidden or Irrelevant files in ```/pipelines``` to avoid breaking imports

### ISSUE THREAD

**Issue by lvxhnat** [kedro-org/kedro#3018]
## Description
I got this issue 

```
[09/08/23 11:47:57] WARNING  /home/yikuang/.local/lib/python3.8/site-packages/kedro/framework/project/__init__.py:359: UserWarning: An error occurred while importing    warnings.py:109
                             the 'sentiomodel.pipelines..ipynb_checkpoints' module. Nothing defined therein will be returned by 'find_pipelines'.                                       
                                                                                                                                                                                        
                             Traceback (most recent call last):                                                                                                                         
                               File "/home/yikuang/.local/lib/python3.8/site-packages/kedro/framework/project/__init__.py", line 357, in find_pipelines                                 
                                 pipeline_module = importlib.import_module(pipeline_module_name)                                                                                        
                               File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module                                                                              
                                 return _bootstrap._gcd_import(name[level:], package, level)                                                                                            
                               File "<frozen importlib._bootstrap>", line 1014, in _gcd_import                                                                                          
                               File "<frozen importlib._bootstrap>", line 991, in _find_and_load                                                                                        
                               File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked                                                                               
                               File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed                                                                             
                               File "<frozen importlib._bootstrap>", line 1014, in _gcd_import                                                                                          
                               File "<frozen importlib._bootstrap>", line 991, in _find_and_load                                                                                        
                               File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked                                                                               
                             ModuleNotFoundError: No module named 'sentiomodel.pipelines.' 
```

## Context
Embarassingly, it did take me a while to figure out what went wrong, since I had configured my vscode to ignore hidden files and so it did not show up on my editor (I had hidden all hidden files but .envs). Turns out the creation of .ipynb_checkpoints has led to faulty import paths, that led to this error.

## Possible Implementation
It will be nice for ```find_pipelines``` to ignore hidden files to prevent this from happening. 

## Possible Alternatives
<!-- (Optional) Describe any alternative solutions or features you've considered. -->

**Comment by astrojuanlu** [comment]
Hi @lvxhnat, thanks for reporting this issue!

Looking at the code, seems like it comes from here:

https://github.com/kedro-org/kedro/blob/e949c6c7d8814ba5fc97c7204c69103a3e9ad21a/kedro/framework/project/__init__.py#L355-L364

**Comment by astrojuanlu** [comment]
Labeling this as a bug

**Comment by lvxhnat** [comment]
Hi @astrojuanlu since the fix doesn't seem to be too big, I have attempted a pull request to fix this issue here #3021 Let me know if this works!

**Comment by astrojuanlu** [comment]
Fixed in #3021
